### DISCUSSION TITLE
Customize LangChain Integration.

### DISCUSSION THREAD

**Discussion by hdp173** [langfuse/langfuse#2146] [Support] [answered]
Hello.

I'm using Langfuse to trace my LangChain llm calls.

But I couldn't find a way to customize Generation name, and prompt.

Generation name is always "ChatOpenAI", but I want to set my name to identify what that completion was for.

Also, I want to connect prompt in langfuse to that generation.

Please help me how to do it.

![image](https://github.com/langfuse/langfuse/assets/35817574/56da9f5e-fa55-4b7e-89e9-d1fbd03ccb2e)

**Comment by marcklingen** [comment]
You can pass a name to your Langchain classes. Have a look here: https://docs.smith.langchain.com/old/cookbook/tracing-examples/runnable-naming

**Reply by hdp173** [reply]
Can I change "ChatOpenAI" to my custom name "IdeaGeneration" with this?

**Reply by hdp173** [reply]
I have multiple generations in one trace and I want to set name for those generations.
![image](https://github.com/langfuse/langfuse/assets/35817574/7a6fa44f-b0e9-4656-a9c3-54fb55bab08e)

**Reply by marcklingen** [reply]
last time i tried this worked, yes

**Reply by hdp173** [reply, marked as answer]
I solved it by setting "name" to ChatOpenAI model params.
Thank you for your support.

**Reply by marcklingen** [reply]
Is this accurate? https://github.com/langfuse/langfuse-docs/blob/main/pages/faq/all/custom-langchain-run-names.mdx

I'd appreciate a note on how you solved this on this page if you have a minute.

**Reply by hdp173** [reply]
If we pass `name` parameter to ChatOpenAI, GENERATION will use that name.
```
trace = langfuse.trace(name="my-trace")
langchain_callbackhandler = trace.get_langchain_handler()
model = ChatOpenAI(name="generate-rating", callbacks=[langchain_callbackhandler])
```
Hope this can be helpful.
Thank you for your support.

![image](https://github.com/langfuse/langfuse/assets/35817574/b0436de7-9917-4d9f-939e-96802815fed0)

**Reply by marcklingen** [reply]
thanks! will add this to the faq page
